code/_helpers/unsorted.dm

```diff
@@ -999,6 +999,91 @@ var/list/wall_items = typecacheof(list(
 			return 1
 	return 0
 
+// Returns a variable type as string, optionally with some details:
+// Objects (datums) get their type, paths get the type name, scalars show length (text) and value (numbers), lists show length.
+// Also attempts some detection of otherwise undetectable types using ref IDs
+var/global/known_proc = new /proc/get_type_ref_bytes
+/proc/get_debug_type(var/V, var/details = TRUE, var/print_numbers = TRUE, var/path_names = TRUE, var/text_lengths = TRUE, var/list_lengths = TRUE, var/show_useless_subtypes = TRUE)
+	// scalars / basic types
+	if(isnull(V))
+		return "null"
+	if(ispath(V))
+		return details && path_names ? "path([V])" : "path"
+	if(istext(V))
+		return details && text_lengths ? "text([length(V) ])" : "text"
+	if(isnum(V)) // Byond doesn't really differentiate between floats and ints, but we can sort of guess here
+		// also technically we could also say that 0 and 1 are boolean but that'd be quite silly
+		if(IsInteger(V) && V < 16777216 && V > -16777216)
+			return details && print_numbers ? "int([V])" : "int"
+		if(V >= INFINITY)
+			return details ? "float(+INF)" : "float"
+		if(V <= -INFINITY)
+			return details ? "float(-INF)" : "float"
+		return details && print_numbers ? "float([V])" : "float"
+	// Resource types
+	if(isicon(V))
+		return "icon"
+	if(isfile(V))
+		return "file"
+	// Types that don't inherit from /datum (note that /world is not here because you can't hold a reference to it)
+	if(istype(V, /list))
```

Comment:
islist is a thing now:
```suggestion
	if(islist(V))
```

Reply:
This would also make special lists like contents/overlays show as list, but idk if that's something you want

Reply:
I'm not sure actually, like it'd be better than showing as unknown, but also should probably be differentiated.

What I'll probably do is keep this as a `istype` check and then do `islist` just before the end.

Reply:
Well I did some experimenting and I can't find anything that would register in `islist` and not with `istype list`, so I'll just use the Byond proc.